mercat/wasm/src/lib.rs

```diff
@@ -0,0 +1,568 @@
+/// * `TransactionJustificationError`: If the mercat library throws an error when creating the proof.
+#[wasm_bindgen]
+pub fn justify_transaction(
+    finalized_tx: Base64,
+    mediator_account: MediatorAccount,
+    sender_public_account: PubAccount,
+    sender_encrypted_pending_balance: Base64,
+    receiver_public_account: PubAccount,
+    ticker_id: String,
+) -> Fallible<JustifiedTransactionOutput> {
+    let mut rng = OsRng;
+
+    let justified_tx = CtxMediator
+        .justify_transaction(
+            decode::<FinalizedTransferTx>(finalized_tx)?,
+            &mediator_account.to_mercat()?.encryption_key,
+            &sender_public_account.to_mercat()?,
+            &decode::<EncryptedAmount>(sender_encrypted_pending_balance)?,
+            &receiver_public_account.to_mercat()?,
+            &[],
+            ticker_id_to_asset_id(ticker_id)?,
+            &mut rng,
+        )
+        .map_err(|_| WasmError::TransactionJustificationError)?;
+
+    Ok(JustifiedTransactionOutput {
+        justified_tx: base64::encode(justified_tx.encode()),
+    })
+}
+
+/// Decrypts an `encrypted_value` given the secret account information.
+///
+/// # Arguments
+/// * `encrypted_value`: The encrypted value.
+/// * `account`: The mercat account. Can be obtained from `CreateAccountOutput.account`.
+///
+/// # Outputs
+/// * `u32`: The decrypted value.
+///
+/// # Errors
+/// * `Base64DecodingError`: If either of the inputs cannot be decoded from base64.
+/// * `DeserializationError`: If either of the inputs cannot be deserialized to a mercat account.
+/// * `DecryptionError`: If the mercat library throws an error while decrypting the value.
+#[wasm_bindgen]
+pub fn decrypt(encrypted_value: Base64, account: Account) -> Fallible<u32> {
+    let enc_balance = decode::<EncryptedAmount>(encrypted_value)?;
+    let account = account.to_mercat()?;
+
+    let decrypted_value = account
+        .secret
+        .enc_keys
+        .secret
+        .decrypt(&enc_balance)
+        .map_err(|_| WasmError::DecryptionError)?;
+
+    Ok(decrypted_value)
+}
+
+// ------------------------------------------------------------------------------------
+// -                               Internal Functions                                 -
+// ------------------------------------------------------------------------------------
+
+fn decode<T: Decode>(data: Base64) -> Fallible<T> {
+    if let Ok(decoded) = base64::decode(data) {
+        if let Ok(ret) = T::decode(&mut &decoded[..]) {
+            return Ok(ret);
+        }
+        return Err(WasmError::DeserializationError.into());
+    }
+
+    Err(WasmError::Base64DecodingError.into())
+}
+
+fn ticker_id_to_asset_id(ticker_id: String) -> Fallible<AssetId> {
+    let mut asset_id = [0u8; 12];
+    let decoded = hex::decode(ticker_id).map_err(|_| WasmError::HexDecodingError)?;
+    asset_id[..decoded.len()].copy_from_slice(&decoded);
```

Comment:
We could save some cycles:
```suggestion
    let asset_id :[u8; TICKER_LEN] = decoded.as_slice().try_into().map_err(|_|  <your_error_code_here> )?;
```